buildtest/cli/build.py

```diff
@@ -310,6 +310,12 @@ def discover_buildspecs_by_tags(buildspec_cache, tagnames):
     buildspecs = []
     # query all buildspecs from BUILDSPEC_CACHE_FILE for tags keyword and
     # if it matches input_tag we add buildspec to list
+    tagnames = [
+        tag.strip() for tagname in tagnames for tag in tagname.split(",") if tag != ""
```

Comment:
i dont like the `if tag != ""` in the list comprehension and it doesnt work all the time

i tried this with a few use-cases where i have spaces between commas and the strip was working

```
>>> tagnames = ['fail, pass', 'python', 'network', " ,  "]
>>> b =  [tag.strip() for tag in tagnames for tag in tag.split(',') if tag.strip()]
>>> b
['fail', 'pass', 'python', 'network']


>>> tagnames = ['fail, pass', 'python', 'network', " , ab ,  "]
>>> b =  [tag.strip() for tag in tagnames for tag in tag.split(',') if tag.strip()]
>>> b
['fail', 'pass', 'python', 'network', 'ab']

```

However this same example with your if condition returns empty characters

```
>>> tagnames = ['fail, pass', 'python', 'network', " , ab ,  "]
>>> b = [ tag.strip() for tagname in tagnames for tag in tagname.split(",") if tag != "" ] 
>>> b
['fail', 'pass', 'python', 'network', '', 'ab', '']

```

Reply:
@shahzebsiddiqui Good catch!